packages/web-app-files/src/views/spaces/GenericSpace.vue

```diff
@@ -90,6 +89,34 @@
             />
           </template>
         </resource-table>
+        <resource-tiles
+          v-else
+          :data="paginatedResources"
+          class="oc-px-m oc-pt-l"
+          @fileClick="$_fileActions_triggerDefaultAction"
+        >
+          <template #actions="{ resource }">
+            <quick-actions
+              :class="resource.preview"
+              class="oc-visible@s"
+              :item="resource"
+              :actions="app.quickActions"
+            />
+          </template>
+          <template #contextMenuActions="{ resource }">
+            <context-actions :space="space" :items="[resource]" />
+          </template>
+          <template #footer>
+            <pagination :pages="paginationPages" :current-page="paginationPage" />
+            <list-info
+              v-if="paginatedResources.length > 0"
+              class="oc-width-1-1 oc-my-s"
+              :files="totalFilesCount.files"
+              :folders="totalFilesCount.folders"
+              :size="totalFilesSize"
+            />
+          </template>
+        </resource-tiles>
```

Comment:
Is the plan to use `component :is='...'` in the long run?

Reply:
Not 100% sure yet tbh, I faced some problems of the `GenericSpace` not handling events being thrown by the switched-out component. Up for discussion I suppose, I can see another possible strategy that's pretty lean code-wise evolve :)